dagshub/data_engine/model/query_result.py

```diff
@@ -208,21 +208,26 @@ def __getitem__(self, item: Union[str, int, slice]):
                 f"needs to be either int or str or slice")
 
     def get_blob_fields(self, *fields: str, load_into_memory=False,
-                        cache_on_disk=True, num_proc: int = config.download_threads) -> "QueryResult":
+                    cache_on_disk=True, num_proc: int = config.download_threads) -> "QueryResult":
         """
         Downloads data from blob fields
 
         Args:
-            fields: list of binary fields to download blobs for
+            fields: list of binary fields to download blobs for. If empty, download all blob fields.
             load_into_memory: Whether to load the blobs into the datapoints, or just store them on disk
-                If True : the datapoints' specified fields will contain the blob data
+                If True: the datapoints' specified fields will contain the blob data
                 If False: the datapoints' specified fields will contain Path objects to the file of the downloaded blob
             cache_on_disk: Whether to cache the blobs on disk or not (valid only if load_into_memory is set to True)
                 Cache location is `~/dagshub/datasets/<user>/<repo>/<datasource_id>/.metadata_blobs/`
         """
         send_analytics_event("Client_DataEngine_downloadBlobs", repo=self.datasource.source.repoApi)
         if not load_into_memory:
             assert cache_on_disk
+
+        # If no fields are specified, include all blob fields from self.datasource.fields
+        if not fields:
+            fields = [field.name for field in self.datasource.fields if field.is_blob()]
```

Comment:
There is no `is_blob()` function on the `MetadataFieldSchema` enum.
You can either add this function, or compare to MetadataFieldType.BLOB in the comprehension

```suggestion
            fields = [field.name for field in self.datasource.fields if field.valueType == MetadataFieldType.BLOB]
```

Don't forget to import MetadataFieldType, I don't think it's imported in QueryResult.